File: app/src/main/java/ai/elimu/soundcards/SoundCardApplication.java
package ai.elimu.soundcards;

import android.app.Application;
import android.speech.tts.TextToSpeech;
import android.util.Log;

public class SoundCardApplication extends Application {

    private TextToSpeech tts;

    @Override
    public void onCreate() {
        Log.i(getClass().getName(), "onCreate");
        super.onCreate();

        // Initialize TTS
        tts = new TextToSpeech(getApplicationContext(), new TextToSpeech.OnInitListener() {
            @Override
            public void onInit(int status) {
                Log.i(getClass().getName(), "TextToSpeech onInit");
                Log.i(getClass().getName(), "TextToSpeech status: " + status);
            }
        });
    }

    public TextToSpeech getTts() {
        return tts;
    }
}

File: app/src/main/java/ai/elimu/soundcards/SoundCardApplication.kt
package ai.elimu.soundcards

import android.app.Application
import android.speech.tts.TextToSpeech
import android.speech.tts.TextToSpeech.OnInitListener
import android.util.Log

class SoundCardApplication : Application() {
    
    private val TAG = "SoundCardApplication"
    var tts: TextToSpeech? = null
        private set

    override fun onCreate() {
        Log.i(TAG, "onCreate")
        super.onCreate()

        // Initialize TTS
        tts = TextToSpeech(applicationContext, object : OnInitListener {
            override fun onInit(status: Int) {
                Log.i(TAG, "TextToSpeech onInit")
                Log.i(TAG, "TextToSpeech status: $status")
            }
        })
    }
}
